support/src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -362,7 +362,11 @@ private static void finish(@CheckForNull String cookie) {
                     return;
                 }
                 final AsynchronousExecution execution = runningTask.execution;
-                assert execution != null && runningTask.launcher != null;
+                if (execution == null) {
```

Comment:
How is this possible? A call to finish on a `node` step when its execution is not even scheduled?